utilities/generate_messages_catalog.py

```diff
@@ -131,96 +134,172 @@ def _is_callable(item):
+    :return: A listing, in dictionary format, of all the messages of the
+        specified module which can then be used to generate the XML list.
+    :rtype: list [dict]
     """
     id_messages = []
-    file_count = 0
-    for python_module in python_package:
-        file_count += 1
-        full_filename_path = os.path.join(package_root, python_module)
-        refFilename = (
-            full_filename_path[len(package_root) + 1:].replace("\\", "/")
-        )
-        with open(full_filename_path, encoding="utf-8") as module_file:
-            tree = ast.parse(module_file.read(), filename=python_module)
-            callables = filter(_is_callable, ast.walk(tree))
-            for item in callables:
-                try:
-                    # imported function could be by id instead of attr
-                    handler = FUNC_HANDLER.get(
-                            item.func.attr, lambda x:  ("", 0)
-                    )
-                    if isinstance(handler, tuple):
-                        level, args_offset = handler
-                    else:
-                        level, args_offset = handler(item)
-
-                    msgCodeArg = item.args[0 + args_offset]  # str or tuple
-                    if isinstance(msgCodeArg,ast.Str):
-                        msgCodes = (msgCodeArg.s,)
+    ref_module_name = os.path.basename(python_module)
+    with open(python_module, encoding="utf-8") as module_file:
+        tree = ast.parse(module_file.read(), filename=python_module)
+        callables = filter(_is_callable, ast.walk(tree))
+        for item in callables:
+            try:
+                # imported function could be by id instead of attr
+                handler = FUNC_HANDLER.get(
+                    item.func.attr, lambda x: ("", 0)
+                )
+                if isinstance(handler, tuple):
+                    level, args_offset = handler
+                else:
+                    level, args_offset = handler(item)
+
+                msgCodeArg = item.args[0 + args_offset]  # str or tuple
+                if isinstance(msgCodeArg,ast.Str):
+                    msgCodes = (msgCodeArg.s,)
+                else:
+                    if any(isinstance(element, (ast.Call, ast.Name))
+                           for element in ast.walk(msgCodeArg)):
+                        msgCodes = ("(dynamic)",)
                     else:
-                        if any(isinstance(element, (ast.Call, ast.Name))
-                               for element in ast.walk(msgCodeArg)):
-                            msgCodes = ("(dynamic)",)
+                        msgCodes = [
+                            element.s for element in ast.walk(msgCodeArg)
+                            if isinstance(element, ast.Str)
+                        ]
+                msgArg = item.args[1 + args_offset]
+                if isinstance(msgArg, ast.Str):
+                    msg = msgArg.s
+                elif ((isinstance(msgArg, ast.Call) and
```

Comment:
This could be extracted out to a single-line function called something like ```_is_translatable``` or something similar, as someone new (or even veteran) to the project might have a hard time figuring out why ```_``` was special.